java/src/jmri/jmrit/decoderdefn/IdentifyDecoder.java

```diff
@@ -293,8 +293,10 @@ public boolean test7(int value) {
             productIDhighest = value;
             if (((productIDlowest >= 129 && productIDlowest <= 135) && (productIDlow == 5))||(modelID >= 5)){
                 productID = productIDlowest+(productIDlow*256)+(productIDhigh*256*256)+(productIDhighest*256*256*256);
-            }
-            else{
+            } else if ((((productIDlowest >= 129 && productIDlowest <= 135) || (productIDlowest >= 170 && productIDlowest <= 172) || productIDlowest == 180) && (modelID == 4)) ||
+                ((productIDlowest == 180) && (modelID == 5))) {
```

Comment:
`modelID` cannot be `5` here. If `modelID` is `5`, the condition on line 294 would be true so this `else if` statement wouldn't be evaluated.

Reply:
yup... fixedit.